macros/generic_tests/at_least_one.sql

```diff
@@ -17,7 +25,14 @@ from (
       {{select_gb_cols}}
       count({{ column_name }}) as filler_column
 
-    from {{ model }}
+    from (
+          select
+            {{select_gb_cols}}
+            {{ column_name }}
+          from {{ model }}
+          where {{ column_name }} is not null
+          limit 1
+          ) pruned_rows
```

Comment:
A third layer of subquery feels pretty ick to me as an adherent to [team CTE](https://discourse.getdbt.com/t/why-the-fishtown-sql-style-guide-uses-so-many-ctes/1091) - could this be pulled out to a `pruned_rows` CTE without impacting performance? If so then that’s my preference.